sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```diff
@@ -885,32 +897,43 @@ def download_manifest(self, repository, tag_or_digest, **kwargs):
         return DownloadManifestResult(digest=digest, data=manifest_stream, manifest=manifest)
 
     @distributed_trace
-    def download_blob(self, repository, digest, **kwargs):
-        # type: (str, str, **Any) -> Union[DownloadBlobResult, None]
-        """Download a blob that is part of an artifact.
+    def download_blob(self, repository: str, digest: str, **kwargs) -> IO:
+        """Download a blob that is part of an artifact to a stream.
 
-        :param str repository: Name of the repository
+        :param str repository: Name of the repository.
         :param str digest: The digest of the blob to download.
-        :returns: DownloadBlobResult or None
-        :rtype: ~azure.containerregistry.DownloadBlobResult or None
+        :returns: IO
+        :rtype: IO
         :raises ValueError: If the parameter repository or digest is None.
         """
+        hasher = hashlib.sha256()
+        downloaded = 0
+        blob_size = None
+        data = BytesIO()
         try:
-            deserialized = self._client.container_registry_blob.get_blob( # type: ignore
-                repository, digest, cls=_return_deserialized, **kwargs
-            )
+            while(True):
```

Comment:
The implementation here only covers the `stream=False` scenario.
We'd also need to implement a `stream=True` scenario as discussed earlier, where the chunks are only downloaded once a user calls `result.data.read()`

Reply:
Oh, I missed this one, is working on it

Reply:
Since we decide to change the api to be always stream the download result now, I'll leave the api as it is until we get further feedback.